include/graphblas/algorithms/triangle_count.hpp

```diff
@@ -0,0 +1,249 @@
+
+/*
+ *   Copyright 2023 Huawei Technologies Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file
+ *
+ * Implements the triangle counting algorithm, using different methods.
+ *
+ * @author Benjamin Lozes
+ * @date: May 10th, 2023
+ */
+
+#ifndef _H_GRB_TRIANGLE_COUNT
+#define _H_GRB_TRIANGLE_COUNT
+
+#include <map>
+#include <numeric>
+#include <vector>
+
+#include <graphblas/utils/iterators/nonzeroIterator.hpp>
+
+#include <graphblas.hpp>
+
+namespace grb {
+
+	namespace algorithms {
+
+		enum class TriangleCountAlgorithm { Burkhardt, Cohen, Sandia_TT };
```

Comment:
doxy comment missing -- may be nice to add the references for each variant in the doxy